[Middleware_01] Built MiddlewareGisInterface Tests and Implementation

jvalentini555 opened the pull request:
Fixes #69

jvalentini555 committed:
[Middleware_01] Define MiddlewareGIS

@brendam19 @martinhughes0722 @BarczakS @duongy18418 @Slackjaw1431

Commit:
Generate SVG and PNG Images for PlantUML Diagrams

jvalentini555 committed:
[Middleware_01] Define Test Method Names

@Bimmerboi-248 @martinhughes0722 @san4497 @Slackjaw1431 @duongy18418

Commit:
Generate SVG and PNG Images for PlantUML Diagrams

jvalentini555 committed:
[Middleware_01] Built MiddlewareGisInterface Classes

Working on #69
Linked to #98

@BarczakS @martinhughes0722 @duongy18418 @shayat97 @Slackjaw1431 @san4497

Commit:
Generate SVG and PNG Images for PlantUML Diagrams

andrewdimmer commented on src/test/java/edu/oakland/test/middleware01/MiddlewareGisInterfaceImplementationTests.java:
```diff
@@ -0,0 +1,53 @@
+import java.lang.IllegalArgumentException;
+import edu.oakland.helper.admin.LocationDataPoint;
+
+@DisplayName("Middleware GIS interface test")
```
Really minor nitpick, but I recommend calling it this so that the logs are consistent across all teams.
```suggestion
@DisplayName("MiddlewareGisInterfaceImplementation Unit Tests")
```

andrewdimmer commented on src/test/java/edu/oakland/test/middleware01/MiddlewareGisInterfaceImplementationTests.java:
```diff
@@ -0,0 +1,53 @@
+@DisplayName("Middleware GIS interface test")
+public class MiddlewareGisInterfaceImplementationTests {
+    
+  @Test
+  @DisplayName("Signal received from Display")
+  void isGpsSignalStrengthReceived() {
+
+  }
+
```
If this test is no longer required, feel free to delete it!
```suggestion
  @Test
  @DisplayName("Signal received from Display")
  void isGpsSignalStrengthReceived() {

  }

```

andrewdimmer commented on src/test/java/edu/oakland/test/middleware01/MiddlewareGisInterfaceImplementationTests.java:
```diff
@@ -0,0 +1,53 @@
+
+  @Test
+  @DisplayName("Signal pass successful from interface to Display")
+
```
Just cleaning up some extra white space... :)
```suggestion
```

andrewdimmer commented on src/test/java/edu/oakland/test/middleware01/MiddlewareGisInterfaceImplementationTests.java:
```diff
@@ -0,0 +1,53 @@
+
+    MiddlewareGisInterface gisInterface = new MiddlewareGisInterfaceImplementation(new MiddlewareGisProcessorStub());
+    assertEquals(satellite.getSatelliteName(), gisInterface.receiveGpsSignalStrengthUc1(satellite));
+
```
Just cleaning up some extra white space... :)
```suggestion
```

andrewdimmer commented on docs/uml-class-diagrams/middleware01/test/MiddlewareGisProcessorImplementationTests/MiddlewareGisProcessorImplementationTests.puml:
```diff
@@ -1,9 +1,11 @@
-
-    }
+  class MiddlewareGisProcessorImplementationTests {
+    + void isGpsSignalStrengthNull()
+    + void checkValueGpsSignalStrength()
```
Just updating the test plan document to match what was implemented.
```suggestion
    + void passSignalStrengthUc1()
    + void passSignalStrengthUc2()
```

jvalentini555 replied:
Isn't what you commented on for the the Processor? We had the Uc1 and Uc2 confusion in the Interface. Though I will change the interface document. @andrewdimmer

andrewdimmer commented on docs/uml-class-diagrams/middleware01/test/MiddlewareGisManagerImplementationTests/MiddlewareGisManagerImplementationTests.puml:
```diff
@@ -1,9 +1,22 @@
 @startuml MiddlewareGisManagerImplementationTests
```
No action required, this test plan looks good, and if it's not already merged feel free to do so. Just for the future though, you may want to make sure that you pull `main` before creating a new branch, and make sure you switch to a new, clean branch after each PR.

andrewdimmer commented on docs/uml-class-diagrams/middleware01/test/MiddlewareGisInterfaceImplementationTests/MiddlewareGisInterfaceImplementationTests.puml:
```diff
@@ -1,9 +1,11 @@
 @startuml MiddlewareGisInterfaceImplementationTests
```
No action required, this test plan looks good, and if it's not already merged feel free to do so. Just for the future though, you may want to make sure that you pull main before creating a new branch, and make sure you switch to a new, clean branch after each PR.

andrewdimmer reviewed (changes requested):
This looks great! A few minor formatting and naming changes, but otherwise this looks perfect to me, and you should be set to merge it once that's all set!

andrewdimmer committed:
Merge branch 'main' into define-MiddlewareGisImplementationTests

andrewdimmer committed:
Generate SVG and PNG Images for PlantUML Diagrams

jvalentini555 committed:
[Middleware_01] Built MiddlewareGisProcessor Tests and Implementation (#103)

* [Middleware_01] Built MiddlewareGisProcessor Tests and Implementation

Created Manager Stub, Processor Implementation and Processor test.

Co-Authored-By: BarczakS <65358794+BarczakS@users.noreply.github.com>
Co-Authored-By: Bimmerboi-248 <73412988+Bimmerboi-248@users.noreply.github.com>
Co-Authored-By: brendam19 <56038943+brendam19@users.noreply.github.com>
Co-Authored-By: duongy18418 <55992627+duongy18418@users.noreply.github.com>
Co-Authored-By: martinhughes0722 <60445081+martinhughes0722@users.noreply.github.com>
Co-Authored-By: san4497 <56036629+san4497@users.noreply.github.com>

* [Middleware__01] Refactored to style guides

Co-Authored-By: BarczakS <65358794+BarczakS@users.noreply.github.com>
Co-Authored-By: Bimmerboi-248 <73412988+Bimmerboi-248@users.noreply.github.com>
Co-Authored-By: brendam19 <56038943+brendam19@users.noreply.github.com>
Co-Authored-By: duongy18418 <55992627+duongy18418@users.noreply.github.com>
Co-Authored-By: martinhughes0722 <60445081+martinhughes0722@users.noreply.github.com>
Co-Authored-By: san4497 <56036629+san4497@users.noreply.github.com>

* [Middleware_01] Placeholder for code that doesn't compile

Left a comment at line 23 in MiddlewareGisProcessorImplementation that stops the error from occuring.

Co-Authored-By: BarczakS <65358794+BarczakS@users.noreply.github.com>
Co-Authored-By: Bimmerboi-248 <73412988+Bimmerboi-248@users.noreply.github.com>
Co-Authored-By: brendam19 <56038943+brendam19@users.noreply.github.com>
Co-Authored-By: duongy18418 <55992627+duongy18418@users.noreply.github.com>
Co-Authored-By: martinhughes0722 <60445081+martinhughes0722@users.noreply.github.com>
Co-Authored-By: san4497 <56036629+san4497@users.noreply.github.com>

* [Middleware_01] Updated Styling

Co-Authored-By: BarczakS <65358794+BarczakS@users.noreply.github.com>
Co-Authored-By: Bimmerboi-248 <73412988+Bimmerboi-248@users.noreply.github.com>
Co-Authored-By: brendam19 <56038943+brendam19@users.noreply.github.com>
Co-Authored-By: duongy18418 <55992627+duongy18418@users.noreply.github.com>
Co-Authored-By: martinhughes0722 <60445081+martinhughes0722@users.noreply.github.com>
Co-Authored-By: san4497 <56036629+san4497@users.noreply.github.com>

* [Middleware_01] Built MiddlewareGISManager Tests and Implementation (#112)

* [Middleware_01] Built MiddlewareGisManager Tests and Implementation

This code as of Nov 28 2020 does not compile because of a dependency on the Satellite strengthIsValid() method.

Co-Authored-By: BarczakS <65358794+BarczakS@users.noreply.github.com>
Co-Authored-By: brendam19 <56038943+brendam19@users.noreply.github.com>
Co-Authored-By: duongy18418 <55992627+duongy18418@users.noreply.github.com>
Co-Authored-By: martinhughes0722 <60445081+martinhughes0722@users.noreply.github.com>

* Update MiddlewareGisManagerImplementationTests.java

Co-Authored-By: BarczakS <65358794+BarczakS@users.noreply.github.com>
Co-Authored-By: brendam19 <56038943+brendam19@users.noreply.github.com>
Co-Authored-By: duongy18418 <55992627+duongy18418@users.noreply.github.com>
Co-Authored-By: martinhughes0722 <60445081+martinhughes0722@users.noreply.github.com>

* [Middleware_01] Fixed styling and added documentation

Co-Authored-By: BarczakS <65358794+BarczakS@users.noreply.github.com>
Co-Authored-By: brendam19 <56038943+brendam19@users.noreply.github.com>
Co-Authored-By: duongy18418 <55992627+duongy18418@users.noreply.github.com>
Co-Authored-By: martinhughes0722 <60445081+martinhughes0722@users.noreply.github.com>

* [Middleware_01] Fixed styling in documentation

Co-Authored-By: BarczakS <65358794+BarczakS@users.noreply.github.com>
Co-Authored-By: brendam19 <56038943+brendam19@users.noreply.github.com>
Co-Authored-By: duongy18418 <55992627+duongy18418@users.noreply.github.com>
Co-Authored-By: martinhughes0722 <60445081+martinhughes0722@users.noreply.github.com>

* Update MiddlewareGisManagerImplementation.java

Co-Authored-By: BarczakS <65358794+BarczakS@users.noreply.github.com>
Co-Authored-By: brendam19 <56038943+brendam19@users.noreply.github.com>
Co-Authored-By: duongy18418 <55992627+duongy18418@users.noreply.github.com>
Co-Authored-By: martinhughes0722 <60445081+martinhughes0722@users.noreply.github.com>

* [Middleware_01] Shortened ManagerImplementationTests line length

Co-Authored-By: BarczakS <65358794+BarczakS@users.noreply.github.com>
Co-Authored-By: brendam19 <56038943+brendam19@users.noreply.github.com>
Co-Authored-By: duongy18418 <55992627+duongy18418@users.noreply.github.com>
Co-Authored-By: martinhughes0722 <60445081+martinhughes0722@users.noreply.github.com>

* [Middleware_01] Implemented changes suggested by Andrew

Co-Authored-By: BarczakS <65358794+BarczakS@users.noreply.github.com>
Co-Authored-By: brendam19 <56038943+brendam19@users.noreply.github.com>
Co-Authored-By: duongy18418 <55992627+duongy18418@users.noreply.github.com>
Co-Authored-By: martinhughes0722 <60445081+martinhughes0722@users.noreply.github.com>

* Generate SVG and PNG Images for PlantUML Diagrams

* Copy and pasted Admin's Satellite class in

Julia did something stupid with the branches and copy and pasting the class in seems to be the easiest way to "fetch" the new class.

* Copy and pasted the other Admin helper classes in

* [Middleware_01] Added private helper method in Processor to determine if Signal is valid

* [Middleware_01] Fixed Manager Test typos

* [Middleware_01] Fixed tests

Co-authored-by: BarczakS <65358794+BarczakS@users.noreply.github.com>
Co-authored-by: brendam19 <56038943+brendam19@users.noreply.github.com>
Co-authored-by: duongy18418 <55992627+duongy18418@users.noreply.github.com>
Co-authored-by: martinhughes0722 <60445081+martinhughes0722@users.noreply.github.com>
Co-authored-by: juliawang2014 <juliawang2014@users.noreply.github.com>

* Generate SVG and PNG Images for PlantUML Diagrams

* [Middleware_01] Added documentation to Implementation Classes

* [Middleware_01] Added a period

Co-authored-by: BarczakS <65358794+BarczakS@users.noreply.github.com>
Co-authored-by: Bimmerboi-248 <73412988+Bimmerboi-248@users.noreply.github.com>
Co-authored-by: brendam19 <56038943+brendam19@users.noreply.github.com>
Co-authored-by: duongy18418 <55992627+duongy18418@users.noreply.github.com>
Co-authored-by: martinhughes0722 <60445081+martinhughes0722@users.noreply.github.com>
Co-authored-by: san4497 <56036629+san4497@users.noreply.github.com>
Co-authored-by: juliawang2014 <juliawang2014@users.noreply.github.com>

jvalentini555 committed:
Update DatabaseGisInterfaceStub.java

jvalentini555 committed:
Merge branch 'main' into define-MiddlewareGisImplementationTests

Commit:
Generate SVG and PNG Images for PlantUML Diagrams

jvalentini555 committed:
[Middleware_01] Fixed Styling

jvalentini555 committed:
[Middleware_01] Shortened line length

jvalentini555 committed:
Delete MiddlewareGisProcessorImplementationTests.java

jvalentini555 committed:
Delete MiddlewareGisProcessorImplementation.java

jvalentini555 committed:
Delete MiddlewareGisManagerStub.java

jvalentini555 committed:
Delete DatabaseGisInterfaceStub.java

jvalentini555 committed:
Delete MiddlewareGisManagerImplementationTests.java

jvalentini555 committed:
Delete MiddlewareGisManagerImplementation.java

andrewdimmer commented on src/main/java/edu/oakland/production/middleware01/MiddlewareGisInterfaceImplementation.java:
```diff
@@ -0,0 +1,45 @@
+package edu.oakland.production.middleware01;
+
+import edu.oakland.helper.admin.Satellite;
+import edu.oakland.production.middleware01.MiddlewareGisInterface;
```
While it's building without it being required, I'd feel better if we imported the `MiddlewareGisProcessor` here as well.
```suggestion
import edu.oakland.production.middleware01.MiddlewareGisInterface;
import edu.oakland.production.middleware01.MiddlewareGisProcessor;
```

andrewdimmer commented on src/main/java/edu/oakland/production/middleware01/MiddlewareGisInterfaceImplementation.java:
```diff
@@ -0,0 +1,45 @@
+    this.middlewareGisProcessor = middlewareGisProcessor;
+    if (this.middlewareGisProcessor == null) {
+      throw new IllegalArgumentException("processor cannot be null");
+    }
```
This is fine if you want to leave it, but generally you want to validate the argument coming in, then only set the attribute if it passes validation. If you want to use that design pattern, it would be like this:
```suggestion
    if (middlewareGisProcessor == null) {
      throw new IllegalArgumentException("processor cannot be null");
    }
    this.middlewareGisProcessor = middlewareGisProcessor;
```

andrewdimmer commented on src/main/java/edu/oakland/production/middleware01/MiddlewareGisInterfaceImplementation.java:
```diff
@@ -0,0 +1,45 @@
+ * This class shall recieve that GPS signal from display and pass it to GIS processor.
+ *
+ */
+
```
Generally you don't want any space between the Javadoc and what it describes, as some IDEs view it differently if there is a space there.
```suggestion
```

andrewdimmer commented on src/test/java/edu/oakland/test/middleware01/MiddlewareGisInterfaceImplementationTests.java:
```diff
@@ -0,0 +1,44 @@
+  @Test
+  @DisplayName("Signal pass successful from interface to Display")
+  void passSignalStrengthUc1() {
+    Satellite sat = new Satellite("GPS0", 5);
```
Generally we don't want to hard code values for testing, and instead want to generate random cases, but this if fine for what you're doing because it is a pass through.

andrewdimmer commented on src/test/java/edu/oakland/test/middleware01/MiddlewareGisInterfaceImplementationTests.java:
```diff
@@ -0,0 +1,44 @@
+  @Test
+  @DisplayName("Signal pass successful from interface to Display")
+  void passSignalStrengthUc2() {
+    Satellite sat = new Satellite("GPS0", 5);
```
Generally we don't want to hard code values for testing, and instead want to generate random cases, but this if fine for what you're doing because it is a pass through.

andrewdimmer reviewed (changes requested):
Couple really, really tiny changes, then you're set to merge!

jvalentini555 committed:
Update src/main/java/edu/oakland/production/middleware01/MiddlewareGisInterfaceImplementation.java

Co-authored-by: Andrew Dimmer <43498863+andrewdimmer@users.noreply.github.com>

jvalentini555 committed:
Update src/main/java/edu/oakland/production/middleware01/MiddlewareGisInterfaceImplementation.java

Co-authored-by: Andrew Dimmer <43498863+andrewdimmer@users.noreply.github.com>

jvalentini555 committed:
Update src/main/java/edu/oakland/production/middleware01/MiddlewareGisInterfaceImplementation.java

Co-authored-by: Andrew Dimmer <43498863+andrewdimmer@users.noreply.github.com>

andrewdimmer reviewed (approved):
You are all set to merge! Well done!
